server/apps/keywords/service.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import logging
+from superdesk import get_resource_service
+from superdesk.errors import SuperdeskApiError
+from superdesk.services import BaseService
+import urllib
+
+from flask import current_app as app
+import requests
+
+
+logger = logging.getLogger(__name__)
+
+
+class KeywordsService(BaseService):
+    def __init__(self, datasource=None, backend=None):
+        super().__init__(datasource, backend)
+        self._http = requests.Session()
+
+    def find_one(self, req, **lookup):
+        service = get_resource_service('archive')
+        doc = service.find_one(req=None, _id=lookup['_id'])
+        if not doc:
+            raise SuperdeskApiError.notFoundError('Can find item content with id: ', lookup['_id'])
+
+        if not app.config['KEYWORDS_KEY_API']:
+            raise SuperdeskApiError.notFoundError('AlchemyAPI key is not set')
+
+        params = {'apikey': app.config['KEYWORDS_KEY_API'],
+                  'outputMode': 'json'}
+
+        url = app.config['KEYWORDS_BASE_URL'] + '/text/TextGetRankedKeywords' + '?' + urllib.parse.urlencode(params)
+        values = {'text': doc.get('body_html', '')}
+        r = self._http.post(url, data=values)
```

Comment:
What happens if the request fails here? IMO such errors should be caught and handled appropriately (e.g. re-raising an error using one of the Superdesk's errors defined for such purpose).